### ISSUE TITLE
Bug: [block-spacing] Block spacing shouldn't affect interfaces, object type literals and enums

### ISSUE THREAD

**Issue by axtk** [typescript-eslint/typescript-eslint#6798]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.2&sourceType=module&code=PTAEDsHtQIwG0gYwNYBoLQKYCduWwM7qTICwAUBXJgC6gAOkAluHQLygDeAHgFygBGdAE9+AWgEBfANwUKIDLAQp0OPIXRQ6JOeRrD6mUAAVmrUBx79wAVwC2MHNNCiI9x9hm7qdRixoC-FZuDk4u1u44XpTkPgyQcMJwLJiBXHwhHs6utqGeANoAuhagRbIxcYyJyeCYAEz8AIK4AIbCADzBuVnhmVEAfCVlQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQEbwHsBjAawFpkAHAQyIEsA7Ac3SgIDN3IAacbSAAIAXAJ6UURaHUpCyKeIyEB6fMXJVajFhkiJo0AtEh8AviBNA

### Repro Code

```typescript
// no block, no errors, ok

let point = {x: 1, y: -1};

// no block, errors, not ok

type Point = {x: number; y: number};

let point1: {x: number; y: number};

let polyline1: {x: number; y: number}[] = [];

let polyline2: Array<{x: number; y: number}> = [];
```

### ESLint Config

```javascript
{
  "rules": {
    "block-spacing": "off",
    "@typescript-eslint/block-spacing": "error"
  }
}
```

### Expected Result

I expect that the block-spacing rule shouldn't report errors in non-blocks such as in `type Point = {x: number; y: number};`, which would be consistent with no errors being reported for `let point = {x: 1, y: -1};`.

### Actual Result

Currently, `type Point = {x: number; y: number};` is marked as having errors requiring extra spaces inside the braces.

### Additional Info

The description to the rule reads:

> This rule extends the base [eslint/block-spacing](https://eslint.org/docs/rules/block-spacing) rule. This version adds support for TypeScript related blocks (interfaces, object type literals and enums).

In fact, interfaces, object type literals and enums are not blocks (as [defined on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/block)) and they shouldn't be added to this rule. Mixing types and blocks together makes it impossible to describe a setting where true blocks are space-padded while object type literals are not.

**Comment by Josh-Cena** [comment]
I agree those are not blocks (enums and interfaces are like class declarations, which are also not blocks). That seems to defeat the purpose of this rule, though—maybe you should just use the base rule?

**Comment by bradzacher** [comment]
The point of the extension is that people wanted these TS constructs treated like blocks because the base rule ignored them.

It sounds like you just want the behaviour from the base rule - so I agree with Josh - if you don't want this extended behaviour, then just use the base rule instead.

**Comment by axtk** [comment]
the base rule does seem to be sufficient (thanks for pointing this out), but there's still a semantic mismatch introduced by the current implementation of the rule; it would be more consistent and less misleading if object type literals, interfaces, and enums were handled with the `object-curly-spacing` rule rather than `block-spacing`

**Comment by JoshuaKGoldberg** [comment]
They're not objects though 😄

**Comment by axtk** [comment]
they describe objects and/or have their shape, it's reasonable to expect they should be formatted the same way too
